### PR TITLE
Add split distribution

### PR DESCRIPTION
**Pull request by NikxDa**
This PR adds support for distributing the remaining balance of a split over all empty splits. This feature was suggested in #1324 and partially implemented in #1271 but never merged. 

**Examples:**

- A transaction has an amount of $200, and three splits each with no amount. Clicking "Distribute" will assign $66.66 to the first two, and $66.67 to the third split, automatically.
- A transaction has an amount of $200, and three splits, two of which have no amount. The third split has an amount of $10. Clicking "Distribute" will assign $95 to the two empty splits.

**Demo:**

CleanShot 2024-01-01 at 17 01 55

As far as I can tell, this is now ready to merge and includes the functionality for both the new transaction editor as well as in existing transactions.

### CODE REVIEW

**Comment by youngcw** [comment]
Nice.  This is working in my quick testing

**Comment by MatissJanis** [comment]
👋 This _feels_ like it's solving a very edge-case problem. Why is the logic to split them equally among the many transactions? The vast majority of purchases (for example: groceries) will not have items with equal value. The amounts will vary drastically. Hence why this _feels_ to me like it's solving an edge-case problem that won't be applicable to most transactions and use-cases. Which is why I would argue against merging this.

However, I would not be against a "fill remaining balance" type button if only a single split transaction remains with empty balance (in which case it's simple to calculate the remaining balance and assign that).

**Comment by NikxDa** [comment]
@MatissJanis I strongly disagree that this is an edge case. Whenever you budget an expense into a reimbursement category, whether it be a set of concert tickets among friends, or a grocery trip with your partner, the amount needs to be split not by individual item but equally. 

*Examples*:
- Grocery shopping with a partner
- Flight tickets for a group of N friends
- Gas for a road trip with N friends

Introducing a "Fill remaining" button seems less useful to me, as the remaining balance is already displayed. Typing that in again is muss less effort than dividing random amounts by two, three, etc.. What makes this especially tedious is that if divisions don't work out, one needs to manually figure out the single cent remainders...

If you really think this isn't a mainstream problem, then can we put this behind a flag in the settings by chance? I know 4 people using Actual including me, and we could all regularly use this.

**Comment by youngcw** [comment]
@NikxDa Can you look into why the test is failing?

**Comment by MatissJanis** [comment]
Thanks. Upon further thought I've reconsidered. Having a button that fills in the remaining amount for the last split transaction (1x) makes sense. And since we'd have that button.. we might as well extend it to do equal splits for multiple split transactions.

**Comment by NikxDa** [comment]
@MatissJanis Thanks, I'm glad to hear it! I can try to make the button text dynamic, as in, if there is only one split left, it will display "Fill", whereas if there are multiple it will display "Distribute" (could make it work so that if all splits are set but don't add up, it displays "Add remaining" and adds the extra split).

@youngcw I will check the tests shortly. Adding splits still works fine, so I'm not sure what's happening there.

**Comment by NikxDa** [comment]
@youngcw Keen to get some input on the failing tests. I fixed a couple, but I can't make sense of the other ones, and I also can't test locally, because my instance database is broken somehow (posted about this on Discord).

**Comment by NikxDa** [comment]
Managed to fix the remaining test. Unclear to me how to fix the visual regression tests, any guidance appreciated.

**Comment by youngcw** [comment]
This explains how to update the visual regression snapshots. https://github.com/actualbudget/actual/blob/master/packages/desktop-client/README.md

**Review by youngcw — APPROVED** [review]

**Comment by NikxDa** [comment]
@youngcw Can you review again, please? I've added some code so that the "Distribute" button is disabled when there are no empty transactions, because it would appear to do nothing in that case.

**Review by youngcw — APPROVED** [review]

File: packages/desktop-client/src/components/transactions/TransactionsTable.jsx [hunk 1]
```diff
@@ -1349,10 +1349,13 @@ const Transaction = memo(function Transaction(props) {
   );
 });
 
-function TransactionError({ error, isDeposit, onAddSplit, style }) {
+function TransactionError({ error, isDeposit, onAddSplit, onDistributeRemainder, style, nestedTransactions }) {
   switch (error.type) {
     case 'SplitTransactionError':
       if (error.version === 1) {
+        const childTransactions = nestedTransactions.filter(t => t.is_child)
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/transactions/TransactionsTable.jsx:1356]: We should just pass a `disabled` prop here and check for amounts of the calling component.

Comment (NikxDa) [reply packages/desktop-client/src/components/transactions/TransactionsTable.jsx:1356]: The reason I didn't opt for this is that the calculation is only necessary if the case is `SplitTransactionError` and the error version is 1. If the calculation is done in the parent components, then it needs to be done for all kinds of errors. Would you still like me to move it?

**Review by joel-jeremy — APPROVED** [review]

**Comment by MattDemers** [comment]
This is cool; I wonder if there could be some kind of option that distributes the remaining amount proportionally to all the split categories, in order to cover a usecase for distributing the subtotal among categories, but having the tax amount left over.

I referenced that [here](https://github.com/actualbudget/actual/issues/1166).

**Pull request merged**